pkg/validate/validate.go

```diff
@@ -15,12 +15,6 @@ func Validate(cs, _ *acsapi.ContainerService) error {
 		return fmt.Errorf("malformed manifest")
 	}
 
-	for _, app := range cs.Properties.AgentPoolProfiles {
-		if app == nil {
-			return fmt.Errorf("malformed manifest")
-		}
-	}
-
```

Comment:
There is len(cs.Properties.AgentPoolProfiles) below which does the same thing. as even empty agentPoolProfile is not nil, so this does not get triggered.

Reply:
not true - cs.Properties.AgentPoolProfiles is a slice of pointers.  This is checking that all the pointers in the slice are non-nil.  It is not checking the length of the slice.
The intention here is to do the non-nil sanity checking once in one place so we don't have to worry about it anywhere else in the codebase (perhaps that's worth a comment - perhaps you could add one?)

Reply:
got u.!